lib/chef/provisioning/docker_driver/docker_transport.rb

```diff
@@ -179,11 +179,12 @@ def upload_file(local_path, path)
     def make_url_available_to_remote(url)
       # The host is already open to the container.  Just find out its address and return it!
       uri = URI(url)
+      uri.scheme = 'http' if 'chefzero' == uri.scheme && uri.host == 'localhost'
```

Comment:
@mwrock the fix we added in https://github.com/chef/chef-provisioning/pull/337 that was released in chef-provisioning 1.2 applies to the _target_ (guest, VM, etc.) node.  It means that chef-client running on that node will be able to talk to the chef-zero server running on the workstation.

This change is different but related.  It makes sure the chef-zero server running on the workstation has its port forwarded to the container.

Having said this, if the user wants to run `chef-client` version 12.3.x or greater on their container they will need to use chef-provisioning ~> 1.2, or use some of the workarounds detailed in https://github.com/chef/chef-provisioning/pull/337.

Actually, I would love to see some documentation about this somewhere.  I've heard multiple people having issues with the changes introduced in chef-client 12.3.  We should at least update https://docs.chef.io/release_notes.html to detail what changed from 12.2 to 12.3 when running `chef-client -z`.